assignment2

```diff
@@ -0,0 +1,171 @@
+public class Node {
+	private int key;
+	private Node leftSon;
+	private Node rightSon;
+
+	public Node(int key) {
+		this.key = key;
+		leftSon = null;
+		rightSon = null;
+	}
+
+	public void insertNode(int key, int parentKey, String position) {
+
+		if (this.key == parentKey) {
+			switch (position) {
+			case "L":
+				if (leftSon == null) {
+					leftSon = new Node(key);
+				}
+				break;
+			case "R":
+				if (rightSon == null) {
+					rightSon = new Node(key);
+				}
+				break;
+			}
+		} else {
+			if (leftSon != null)
+				leftSon.insertNode(key, parentKey, position);
+			if (rightSon != null)
+				rightSon.insertNode(key, parentKey, position);
+		}
+	}
+	
+	public void printTree() {
+		System.out.print(key + " ( ");
+		
+		if (leftSon != null)
+			leftSon.printTree();
+		else
+			System.out.print("- ");
+		if (rightSon != null)
+			rightSon.printTree();
+		else
+			System.out.print("- )");
+		System.out.print(" )");
+	}
+
+	public int getKey() {
+		return key;
+	}
+
+	public void setKey(int key) {
+		this.key = key;
+	}
+
+	public Node getLeftSon() {
+		return leftSon;
+	}
+
+	public void setLeftSon(Node leftSon) {
+		this.leftSon = leftSon;
+	}
+
+	public Node getRightSon() {
+		return rightSon;
+	}
+
+	public void setRightSon(Node rightSon) {
+		this.rightSon = rightSon;
+	}
+}
+
+
+
+import java.util.Scanner;
+import java.util.Stack;
+
+public class Main {
+
+	/******************** Question 2 ********************/
+	public static Node lowestCommonAncestor(int key1, int key2, Node root) {
+		if (root == null)
+			return null;
+
+		if (root.getKey() == key1 || root.getKey() == key2)
+			return root;
+
+		Node left = lowestCommonAncestor(key1, key2, root.getLeftSon());
+		Node right = lowestCommonAncestor(key1, key2, root.getRightSon());
+
+		if (left != null && right != null)
+			return root;
+		else if (left != null)
+			return left;
+		else
+			return right;
+	}
+	
+	
+	/******************** Question 1 ********************/
+	public static Boolean auxQ1(Stack<Node> parents, int key, Node root) {
```

Comment:
what does the name if this method stand for? Could you rename it so that it says what it does?